### DISCUSSION TITLE
Using social auth exclusively

### DISCUSSION THREAD

**Discussion by CrackHead007** [pallets-eco/flask-security#818] [Q&A] [answered]
Is there any way to make flask-security only allow login via social auth.

Registration would be done similar to what is mentioned in this issue: https://github.com/Flask-Middleware/flask-security/issues/751
So there wouldn't really be any password in the database, and would need to set the login url directly to the oauthstart route.

How could this be implemented?

**Comment by jwag956** [comment, marked as answer]
This should be doable since FS already (using unified signin feature) doesn't require a stored password.
I would ask first however - what feature(s) of FS do you actually want to use? if you only authenticate with social auth, and register with social auth - seems like the only feature left is two-factor.. is that what you are using FS for?

If you are using two-factor - then you have to register the FS blueprint - (which has /login) but no one will be able to log in with that since they don't have a password - so you can simply have your own /mylogin endpoint that redirects to the OAUTH_START_URL.

**Reply by CrackHead007** [reply]
FS is already integrated in our codebase we are just trying to change over to SSO from the original setup of normal user management.
I guess we would continue using FS for the integration with Flask-Principal (And continue using the existing role/permission management that we already have setup).
Any examples in general of setting up OAuth with FS, as the documentation seems to be quite sparse.

Thanks a lot
